src/Opaleye/Internal/Column.hs

```diff
@@ -10,12 +15,14 @@ import qualified Opaleye.Internal.HaskellDB.PrimQuery as HPQ
 --
 -- Do not use the 'Show' instance of 'Column'.  It will be deprecated
 -- in version 0.6.
-newtype Column pgType = Column HPQ.PrimExpr deriving Show
+data Column_ colType pgType = Column HPQ.PrimExpr deriving Show
+type Column = Column_ NonNullable
 
 -- | Only used within a 'Column', to indicate that it can be @NULL@.
 -- For example, a 'Column' ('Nullable' @PGText@) can be @NULL@ but a
 -- 'Column' @PGText@ cannot.
 data Nullable a = Nullable
+data NonNullable
```

Comment:
No we cannot write `Column Nullable b` and `Column NonNullable b` when `Nullable` and `NonNullable` have different kinds, but I didn't think that is what this intermediate step was about. My understanding was we are simply laying the foundation for a change that will be abrupt in a future release, but you seem to have a different expectation.

Can you clarify your expectation for this step?

I don't understand the value of `NonNullable` and `Nullable` with kind `* -> *` since that would allow even more nonsensical things, such as `Nullable (NonNullable a)`